pkg/trace/remoteconfighandler/remote_config_handler_test.go

```diff
@@ -0,0 +1,207 @@
+package remoteconfighandler
+
+import (
+	"testing"
+
+	"github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
+	"github.com/DataDog/datadog-agent/pkg/remoteconfig/state/products/apmsampling"
+	"github.com/DataDog/datadog-agent/pkg/trace/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/sampler"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestStart(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	remoteClient := NewMockRemoteClient(ctrl)
+	agentConfig := config.AgentConfig{RemoteSamplingClient: remoteClient}
+	prioritySampler := NewMockprioritySampler(ctrl)
+	errorsSampler := NewMockerrorsSampler(ctrl)
+	rareSampler := NewMockrareSampler(ctrl)
+
+	h := New(&agentConfig, prioritySampler, rareSampler, errorsSampler)
+
+	remoteClient.EXPECT().RegisterAPMUpdate(gomock.Any()).Times(1)
+	remoteClient.EXPECT().Start().Times(1)
+
+	h.Start()
+
+	ctrl.Finish()
+}
+
+func TestStartNoRemoteClient(t *testing.T) {
+	var h *RemoteConfigHandler
+	assert.NotPanics(t, h.Start)
+}
+
+func TestRemoteRates(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	remoteClient := NewMockRemoteClient(ctrl)
+	agentConfig := config.AgentConfig{RemoteSamplingClient: remoteClient}
+	prioritySampler := NewMockprioritySampler(ctrl)
+	errorsSampler := NewMockerrorsSampler(ctrl)
+	rareSampler := NewMockrareSampler(ctrl)
+
+	h := New(&agentConfig, prioritySampler, rareSampler, errorsSampler)
+
+	payload := apmsampling.APMSampling{
+		TargetTPS: []apmsampling.TargetTPS{{
+			Service:   "service-1",
+			Env:       "env-1",
+			Value:     42,
+			Rank:      1,
+			Mechanism: 1,
+		}},
+	}
+
+	raw, _ := payload.MarshalMsg(nil)
+
+	remoteRatesConfig := state.APMSamplingConfig{
+		Metadata: state.Metadata{Version: 42},
+		Config:   raw,
+	}
+
+	prioritySampler.EXPECT().UpdateRemoteRates([]sampler.RemoteRateUpdate{{Version: 42, Config: payload}}).Times(1)
+
+	h.onUpdate(map[string]state.APMSamplingConfig{"datadog/2/APM_SAMPLING/dynamic_rates/config": remoteRatesConfig})
+
+	ctrl.Finish()
+}
+
+func TestPrioritySampler(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	remoteClient := NewMockRemoteClient(ctrl)
+	prioritySampler := NewMockprioritySampler(ctrl)
+	errorsSampler := NewMockerrorsSampler(ctrl)
+	rareSampler := NewMockrareSampler(ctrl)
+
+	agentConfig := config.AgentConfig{RemoteSamplingClient: remoteClient, TargetTPS: 41, ErrorTPS: 41, RareSamplerEnabled: true}
+	h := New(&agentConfig, prioritySampler, rareSampler, errorsSampler)
+
+	payload := apmsampling.SamplerConfig{
+		AllEnvs: apmsampling.SamplerEnvConfig{
+			PrioritySamplerTargetTPS: floatPointer(42),
+		},
+	}
+
+	raw, _ := payload.MarshalMsg(nil)
+	config := state.APMSamplingConfig{
+		Config: raw,
+	}
+
+	prioritySampler.EXPECT().UpdateTargetTPS(*floatPointer(42)).Times(1)
```

Comment:
Do you need this `floatPointer` helper? It seems like it should be safe to just say `float64(42)`? I could be wrong here though as I've never used `gomock` before

Reply:
it totally works, I still need to practice writing Go 😁